frappe/integrations/doctype/ldap_settings/ldap_settings.py

```diff
@@ -142,6 +148,61 @@ def get_ldap_attributes(self):
 
 		return ldap_attributes
 
+
+	def fetch_ldap_groups(self, user, conn):
+		fetch_ldap_groups = None
```

Comment:
Refactor:

Validate `user` to confirm `type()` = `frappe.document.user`

Validate `conn` to confirm `type()` = `ldap3.Connection`

Is there a frappe standard invalid exception?

Reply:
Python has within BaseException the following for wrong types.

REF: https://docs.python.org/3/library/exceptions.html#TypeError

``` python
raise TypeError("display msg")
```

Reply:
implemented in commit 99b141017fa587e60fd13aa21a1017ab6c42806c